stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -435,59 +435,38 @@ internal class ChannelControllerImpl(
-        val queryOnlineJob = if (domainImpl.isOnline()) {
-            domainImpl.scope.async { runChannelQueryOnline(pagination) }
-        } else {
-            // if we are not offline we mark it as needing recovery
-            recoveryNeeded = true
-            logger.logI("Skipping channel.watch for channel $cid since we are offline. Marking it as needing recovery.")
-            null
-        }
-        val localChannel = queryOfflineJob.await()
-        if (localChannel != null) {
+        val queryOnlineJob = domainImpl.scope.async { runChannelQueryOnline(pagination) }
+        val localChannel = queryOfflineJob.await()?.also {
             if (pagination.messageFilterDirection == Pagination.LESS_THAN) {
-                updateOldMessagesFromLocalChannel(localChannel)
+                updateOldMessagesFromLocalChannel(it)
             } else {
-                updateLiveDataFromLocalChannel(localChannel)
+                updateLiveDataFromLocalChannel(it)
             }
+            loader.value = false
         }
 
-        // if we are online we we run the actual API call
-        val result = if (queryOnlineJob != null) {
-            val response = queryOnlineJob.await()
-            if (response.isSuccess) {
-                updateLiveDataFromChannel(response.data())
+        val result: Result<Channel> = queryOnlineJob.await().let { onlineResult ->
+            if (onlineResult.isSuccess) {
+                onlineResult.also { updateLiveDataFromChannel(onlineResult.data()) }
             } else {
-                if (response.error().isPermanent()) {
-                    logger.logW("Permanent failure calling channel.watch for channel $cid, with error ${response.error()}")
+                if (onlineResult.error().isPermanent()) {
+                    logger.logW("Permanent failure calling channel.watch for channel $cid, with error ${onlineResult.error()}")
                 } else {
-                    logger.logW("Temporary failure calling channel.watch for channel $cid. Marking the channel as needing recovery. Error was ${response.error()}")
+                    logger.logW("Temporary failure calling channel.watch for channel $cid. Marking the channel as needing recovery. Error was ${onlineResult.error()}")
                     recoveryNeeded = true
                 }
-            }
-            response
-        } else {
-            if (localChannel != null) {
-                Result(localChannel)
-            } else {
-                Result(ChatError("Local channel was not found"))
+                localChannel?.let { Result(it) } ?: onlineResult
             }
         }
         loader.value = false
         return result
     }
 
-    suspend fun runChannelQueryOffline(pagination: QueryChannelPaginationRequest): Channel? {
-        val selectedChannel = domainImpl.selectAndEnrichChannel(cid, pagination)
-
-        selectedChannel?.also { channel ->
-            channel.config = domainImpl.getChannelConfig(channel.type)
```

Comment:
Do we not need this line that updates the channel config anymore?

Reply:
`domainImpl.selectAndEnrichChannel()` already inject `config` into the returned channel, this call was redundant

Reply:
Nice catch 👍